---
title: "Hawaii2025_dwv_recomb_working"
author: "Laura"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Project

Here I am exploring the data from Oxford Nanopore seq data from pooled bee samples collected from Hawaii in 2024/2025, we are looking at the viral lanscape, particularly DWV in Hawaii 15 years after our first study.

There are 3 islands with different conditions: Oahu - treatment free, Big Island - treat for Varroa, and Kauai - no Varroa.

RNA was extracted in Declan's lab with Clarissa and run across 4 libraries on the GridION

These are the mapping data from mapping to a custom db with minimap2 in epi2me labs. Here I will visualise coverage plots across DWV-A and DWV-B genomes to identify recombinants.

I am then combining this with the data from previous years, to look at the timeline. The older samples are done using illumina seq so the plots look a bit different



### install packages

```{r}

#if (!require("BiocManager", quietly = TRUE))
#    install.packages("BiocManager")

#BiocManager::install("Rsamtools")
#BiocManager::install("gmoviz")

library(Rsamtools) # for reading in the bam files
library(gmoviz) # for getCoverage
library(ggplot2)
library(dplyr)
library(patchwork)


```


## first, the 2025 data


First extract the relevant portions of the alignment:

 "NC_004830.2" = DWV-A
 "NC_006494.1" = DWV-B
 
Then covert to data frames, combine them and plot



### Big Island 2025 - actually these were collected in 2024, as were Oahu

Load the data. First copy all the relevant  bam files to a folder all together. Again, this folder must contain all .bam and corresponding .bam.bai files. Make a loop to collate all the BI data

```{r}

# Define the folder containing BAM files
bam_folder <- "./bams_BI/"

# Get a list of all BAM files in the folder
bam_files <- list.files(path = bam_folder, pattern = "\\.bam$", full.names = TRUE)

# Define chromosomes of interest. These are viruses not chromosomes, just typically this is done with chromosomes so follow chromosome advice
chroms_of_interest <- c("NC_004830.2", "NC_006494.1")

# Initialize an empty list to store data frames
all_coverage_data <- list()

# Loop through each BAM file and extract coverage data
for (bamfile in bam_files) {
  
  # Extract file name without path and extension
  bam_name <- tools::file_path_sans_ext(basename(bamfile))
  
  # Loop through chromosomes and extract coverage data
  for (chrom in chroms_of_interest) {
    temp <- getCoverage(regions_of_interest = chrom, bam_file = bamfile, window_size = 20)
    
    # Convert to data frame and add columns for BAM file and chromosome
    coverage_df <- as.data.frame(temp) %>%
      mutate(chromosome = chrom, bam = bam_name)
    
    # Append to the list
    all_coverage_data[[paste0(bam_name, "_", chrom)]] <- coverage_df
  }
}

# Combine all data frames into one big data frame
coverage_df_BI_2025 <- bind_rows(all_coverage_data)

# View the first few rows of the combined data
head(coverage_df_BI_2025)

```

BI plots and recomb junctions

```{r}
# Bin the data and calculate mean coverage per virus per bin:

bin_width <- 200  # the resolution that worked well after trying a few

bin_summary <- coverage_df_BI_2025 %>%
  mutate(bin = floor(start / bin_width) * bin_width) %>%
  group_by(bam, bin, chromosome) %>%
  summarize(mean_cov = mean(coverage, na.rm = TRUE), .groups = "drop")

# For each bin (200bp section), define the virus with higher mean coverage:

bin_dominance <- bin_summary %>%
  group_by(bam, bin) %>%
  slice_max(mean_cov, n = 1, with_ties = FALSE) %>%  # pick the dominant virus
  ungroup()


# Identify switch points in dominance (between bins), from a switch point we will infer an approximate recombination junction : 

switch_df <- bin_dominance %>%
  group_by(bam) %>%
  arrange(bin) %>%
  mutate(prev_chrom = lag(chromosome)) %>%
  filter(chromosome != prev_chrom & !is.na(prev_chrom)) %>%
  select(bam, bin) %>%
  rename(start_pos = bin)


# make coverage plots with vertical lines to denote approx junction positions

plot_BI_2025_label <- ggplot(coverage_df_BI_2025, aes(x = start, y = coverage, color = chromosome)) +
  geom_line() +
  facet_wrap(~bam, scales = "free_y", ncol = 4) +
  theme_minimal() +
  labs(x = "Genomic Position", y = "Read Depth") +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) +
  scale_color_manual(values = c("red", "blue")) +
  ggtitle("Big island 2025") +
  # Add vertical lines where coverage dominance flips
  geom_vline(data = switch_df, aes(xintercept = start_pos), color = "black", linetype = "dashed")

plot_BI_2025_label



# extra

switch_df_labeled_BI2025 <- bin_dominance %>%
  group_by(bam) %>%
  arrange(bin) %>%
  mutate(prev_chrom = lag(chromosome)) %>%
  filter(chromosome != prev_chrom & !is.na(prev_chrom)) %>%
  rename(start_pos = bin) %>%
  select(bam, start_pos, chromosome)

#ggsave("27Jun_Big_Island_DWV_coverage_label_2.pdf", plot = plot_BI_2025_label, width = 7, height = 4)

```


### Oahu

same process as big island

```{r}


bam_folder_o <- "./bams_oahu/" 

bam_files_o <- list.files(path = bam_folder_o, pattern = "\\.bam$", full.names = TRUE)

chroms_of_interest <- c("NC_004830.2", "NC_006494.1")

all_coverage_data_o <- list()

for (bamfile_o in bam_files_o) {
  
    bam_name_o <- tools::file_path_sans_ext(basename(bamfile_o))
  
    for (chrom in chroms_of_interest) {
    temp <- getCoverage(regions_of_interest = chrom, bam_file = bamfile_o, window_size = 20)
    
      coverage_df_o <- as.data.frame(temp) %>%
      mutate(chromosome = chrom, bam = bam_name_o)
    
      all_coverage_data_o[[paste0(bam_name_o, "_", chrom)]] <- coverage_df_o
  }
}

coverage_df_O_2025 <- bind_rows(all_coverage_data_o)

head(coverage_df_O_2025)

```



Oahu 2025 plots


```{r}
# forthe bin dominance commands, just overwriting objects from the original BI ones, then saving the plot and label df separately for saving

bin_summary <- coverage_df_O_2025 %>%
  mutate(bin = floor(start / bin_width) * bin_width) %>%
  group_by(bam, bin, chromosome) %>%
  summarize(mean_cov = mean(coverage, na.rm = TRUE), .groups = "drop")


bin_dominance <- bin_summary %>%
  group_by(bam, bin) %>%
  slice_max(mean_cov, n = 1, with_ties = FALSE) %>%  
  ungroup()



switch_df <- bin_dominance %>%
  group_by(bam) %>%
  arrange(bin) %>%
  mutate(prev_chrom = lag(chromosome)) %>%
  filter(chromosome != prev_chrom & !is.na(prev_chrom)) %>%
  select(bam, bin) %>%
  rename(start_pos = bin)


plot_O_2025_label <- ggplot(coverage_df_O_2025, aes(x = start, y = coverage, color = chromosome)) +
  geom_line() +
  facet_wrap(~bam, scales = "free_y", ncol = 4) +
  theme_minimal() +
  labs(x = "Genomic Position", y = "Read Depth") +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) +
  scale_color_manual(values = c("red", "blue")) +
  ggtitle("Oahu 2025") +
   geom_vline(data = switch_df, aes(xintercept = start_pos), color = "black", linetype = "dashed")

plot_O_2025_label



# extra

switch_df_labeled_O2025 <- bin_dominance %>%
  group_by(bam) %>%
  arrange(bin) %>%
  mutate(prev_chrom = lag(chromosome)) %>%
  filter(chromosome != prev_chrom & !is.na(prev_chrom)) %>%
  rename(start_pos = bin) %>%
  select(bam, start_pos, chromosome)



#ggsave("27Jun_Oahu_DWV_coverage_label.pdf", plot = plot_O_2025_label, width = 12, height = 14)

```

## now for the older data

## Oahu 2009

same process as before

```{r}

bam_folder_2009 <- "../data_old_runs/Old_Hawaii_data_reanalysis2025/bams_2009/"

bam_files_2009 <- list.files(path = bam_folder_2009, pattern = "\\.bam$", full.names = TRUE)

all_coverage_data_2009 <- list()

for (bamfile_2009 in bam_files_2009) {
  
  bam_name_2009 <- tools::file_path_sans_ext(basename(bamfile_2009))
  
  for (chrom in chroms_of_interest) {
    temp <- getCoverage(regions_of_interest = chrom, bam_file = bamfile_2009, window_size = 20)
    

    coverage_df_2009 <- as.data.frame(temp) %>%
      mutate(chromosome = chrom, bam = bam_name_2009)

    all_coverage_data_2009[[paste0(bam_name_2009, "_", chrom)]] <- coverage_df_2009
  }
}

# I still get the error 'the index file is older than the bam file for everything, but that's fine - that's because I indexed second 

# Combine all data frames into one big data frame
coverage_df_o_2009 <- bind_rows(all_coverage_data_2009)

# write.csv(coverage_df_all, "DWV_coverage_df_all.csv") # save this, then can run using this input rather than all the bams if need be

# View the first few rows of the combined data
head(coverage_df_o_2009)

```


## Oahu 2012

```{r}

# Define the folder containing BAM files
bam_folder <- "../data_old_runs/Old_Hawaii_data_reanalysis2025/bams_2012/bams_2012_o/"

# Get a list of all BAM files in the folder
bam_files_o_2012 <- list.files(path = bam_folder, pattern = "\\.bam$", full.names = TRUE)

# Initialize an empty list to store data frames
all_coverage_data_o_2012 <- list()

# Loop through each BAM file and extract coverage data
for (bamfile_o_2012 in bam_files_o_2012) {
  
  # Extract file name without path and extension
  bam_name_o_2012 <- tools::file_path_sans_ext(basename(bamfile_o_2012))
  
  # Loop through chromosomes and extract coverage data
  for (chrom in chroms_of_interest) {
    temp <- getCoverage(regions_of_interest = chrom, bam_file = bamfile_o_2012, window_size = 20)
    
    # Convert to data frame and add columns for BAM file and chromosome
    coverage_df_o_2012 <- as.data.frame(temp) %>%
      mutate(chromosome = chrom, bam = bam_name_o_2012)
    
    # Append to the list
    all_coverage_data_o_2012[[paste0(bam_name_o_2012, "_", chrom)]] <- coverage_df_o_2012
  }
}



# Combine all data frames into one big data frame
coverage_df_o_2012 <- bind_rows(all_coverage_data_o_2012)

# write.csv(coverage_df_all, "DWV_coverage_df_all.csv") # save this, then can run using this input rather than all the bams if need be

# View the first few rows of the combined data
head(coverage_df_o_2012)

```

## BI 2012

```{r}

# Define the folder containing BAM files
bam_folder <- "../data_old_runs/Old_Hawaii_data_reanalysis2025/bams_2012/bams_2012_BI/"

# Get a list of all BAM files in the folder
bam_files_BI_2012 <- list.files(path = bam_folder, pattern = "\\.bam$", full.names = TRUE)

# Initialize an empty list to store data frames
all_coverage_data_BI_2012 <- list()

# Loop through each BAM file and extract coverage data
for (bamfile_BI_2012 in bam_files_BI_2012) {
  
  # Extract file name without path and extension
  bam_name_BI_2012 <- tools::file_path_sans_ext(basename(bamfile_BI_2012))
  
  # Loop through chromosomes and extract coverage data
  for (chrom in chroms_of_interest) {
    temp <- getCoverage(regions_of_interest = chrom, bam_file = bamfile_BI_2012, window_size = 20)
    
    # Convert to data frame and add columns for BAM file and chromosome
    coverage_df_BI_2012 <- as.data.frame(temp) %>%
      mutate(chromosome = chrom, bam = bam_name_BI_2012)
    
    # Append to the list
    all_coverage_data_BI_2012[[paste0(bam_name_BI_2012, "_", chrom)]] <- coverage_df_BI_2012
  }
}



# Combine all data frames into one big data frame
coverage_df_BI_2012 <- bind_rows(all_coverage_data_BI_2012)

# write.csv(coverage_df_all, "DWV_coverage_df_all.csv") # save this, then can run using this input rather than all the bams if need be

# View the first few rows of the combined data
head(coverage_df_BI_2012)

```
## Oahu 2015

```{r}

# Define the folder containing BAM files
bam_folder <- "../data_old_runs/Old_Hawaii_data_reanalysis2025/bams_2015/bams_2015_o/"

# Get a list of all BAM files in the folder
bam_files_o_2015 <- list.files(path = bam_folder, pattern = "\\.bam$", full.names = TRUE)

# Initialize an empty list to store data frames
all_coverage_data_o_2015 <- list()

# Loop through each BAM file and extract coverage data
for (bamfile_o_2015 in bam_files_o_2015) {
  
  # Extract file name without path and extension
  bam_name_o_2015 <- tools::file_path_sans_ext(basename(bamfile_o_2015))
  
  # Loop through chromosomes and extract coverage data
  for (chrom in chroms_of_interest) {
    temp <- getCoverage(regions_of_interest = chrom, bam_file = bamfile_o_2015, window_size = 20)
    
    # Convert to data frame and add columns for BAM file and chromosome
    coverage_df_o_2015 <- as.data.frame(temp) %>%
      mutate(chromosome = chrom, bam = bam_name_o_2015)
    
    # Append to the list
    all_coverage_data_o_2015[[paste0(bam_name_o_2015, "_", chrom)]] <- coverage_df_o_2015
  }
}



# Combine all data frames into one big data frame
coverage_df_o_2015 <- bind_rows(all_coverage_data_o_2015)

# write.csv(coverage_df_all, "DWV_coverage_df_all.csv") # save this, then can run using this input rather than all the bams if need be

# View the first few rows of the combined data
head(coverage_df_o_2015)

```

## BI 2015

```{r}

# Define the folder containing BAM files
bam_folder <- "../data_old_runs/Old_Hawaii_data_reanalysis2025/bams_2015/bams_2015_BI/"

# Get a list of all BAM files in the folder
bam_files_BI_2015 <- list.files(path = bam_folder, pattern = "\\.bam$", full.names = TRUE)

# Initialize an empty list to store data frames
all_coverage_data_BI_2015 <- list()

# Loop through each BAM file and extract coverage data
for (bamfile_BI_2015 in bam_files_BI_2015) {
  
  # Extract file name without path and extension
  bam_name_BI_2015 <- tools::file_path_sans_ext(basename(bamfile_BI_2015))
  
  # Loop through chromosomes and extract coverage data
  for (chrom in chroms_of_interest) {
    temp <- getCoverage(regions_of_interest = chrom, bam_file = bamfile_BI_2015, window_size = 20)
    
    # Convert to data frame and add columns for BAM file and chromosome
    coverage_df_BI_2015 <- as.data.frame(temp) %>%
      mutate(chromosome = chrom, bam = bam_name_BI_2015)
    
    # Append to the list
    all_coverage_data_BI_2015[[paste0(bam_name_BI_2015, "_", chrom)]] <- coverage_df_BI_2015
  }
}



# Combine all data frames into one big data frame
coverage_df_BI_2015 <- bind_rows(all_coverage_data_BI_2015)

# write.csv(coverage_df_all, "DWV_coverage_df_all.csv") # save this, then can run using this input rather than all the bams if need be

# View the first few rows of the combined data
head(coverage_df_BI_2015)

```

## Oahu 2016

```{r}

# Define the folder containing BAM files
bam_folder <- "../data_old_runs/Old_Hawaii_data_reanalysis2025/bams_2016"

# Get a list of all BAM files in the folder
bam_files_o_2016 <- list.files(path = bam_folder, pattern = "\\.bam$", full.names = TRUE)

# Initialize an empty list to store data frames
all_coverage_data_o_2016 <- list()

# Loop through each BAM file and extract coverage data
for (bamfile_o_2016 in bam_files_o_2016) {
  
  # Extract file name without path and extension
  bam_name_o_2016 <- tools::file_path_sans_ext(basename(bamfile_o_2016))
  
  # Loop through chromosomes and extract coverage data
  for (chrom in chroms_of_interest) {
    temp <- getCoverage(regions_of_interest = chrom, bam_file = bamfile_o_2016, window_size = 20)
    
    # Convert to data frame and add columns for BAM file and chromosome
    coverage_df_o_2016 <- as.data.frame(temp) %>%
      mutate(chromosome = chrom, bam = bam_name_o_2016)
    
    # Append to the list
    all_coverage_data_o_2016[[paste0(bam_name_o_2016, "_", chrom)]] <- coverage_df_o_2016
  }
}



# Combine all data frames into one big data frame
coverage_df_o_2016 <- bind_rows(all_coverage_data_o_2016)

# write.csv(coverage_df_all, "DWV_coverage_df_all.csv") # save this, then can run using this input rather than all the bams if need be

# View the first few rows of the combined data
head(coverage_df_o_2016)

```

## oahu plots 


Now with labels (and new df)

```{r}

# 2009

# again will overwrite the bin dominance objects

bin_summary <- coverage_df_o_2009 %>%
  mutate(bin = floor(start / bin_width) * bin_width) %>%
  group_by(bam, bin, chromosome) %>%
  summarize(mean_cov = mean(coverage, na.rm = TRUE), .groups = "drop")


bin_dominance <- bin_summary %>%
  group_by(bam, bin) %>%
  slice_max(mean_cov, n = 1, with_ties = FALSE) %>%  
  ungroup()


switch_df <- bin_dominance %>%
  group_by(bam) %>%
  arrange(bin) %>%
  mutate(prev_chrom = lag(chromosome)) %>%
  filter(chromosome != prev_chrom & !is.na(prev_chrom)) %>%
  select(bam, bin) %>%
  rename(start_pos = bin)



plot_o_2009 <- ggplot(coverage_df_o_2009, aes(x = start, y = coverage, color = chromosome)) +
    geom_point(alpha = 0.4, size = 1) +
  #geom_smooth(method = "loess", span = 0.2, se = FALSE) +  # Apply smoothing
    facet_wrap(~bam, scales = "free_y", ncol = 4) + # free_y makes all plots have logical y axes, without this they are all plotted on the same axis scale and some are too small to see properly
    theme_minimal() +
    labs(x = "Genomic Position",
         y = "Read Depth (log10 scale)") +
         theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
               aspect.ratio = 1) +  # Ensures all facets have the same shape) + 
    scale_color_manual(values = c("red", "blue"))+
    scale_y_log10() + 
    ggtitle("2009")+
  theme(legend.position = "none") +
  geom_vline(data = switch_df, aes(xintercept = start_pos), color = "black", linetype = "dashed")


#plot_o_2009




# make df

switch_df_labeled_O2009 <- bin_dominance %>%
  group_by(bam) %>%
  arrange(bin) %>%
  mutate(prev_chrom = lag(chromosome)) %>%
  filter(chromosome != prev_chrom & !is.na(prev_chrom)) %>%
  rename(start_pos = bin) %>%
  select(bam, start_pos, chromosome)





# 2012

bin_summary <- coverage_df_o_2012 %>%
  mutate(bin = floor(start / bin_width) * bin_width) %>%
  group_by(bam, bin, chromosome) %>%
  summarize(mean_cov = mean(coverage, na.rm = TRUE), .groups = "drop")

bin_dominance <- bin_summary %>%
  group_by(bam, bin) %>%
  slice_max(mean_cov, n = 1, with_ties = FALSE) %>%  
  ungroup()

switch_df <- bin_dominance %>%
  group_by(bam) %>%
  arrange(bin) %>%
  mutate(prev_chrom = lag(chromosome)) %>%
  filter(chromosome != prev_chrom & !is.na(prev_chrom)) %>%
  select(bam, bin) %>%
  rename(start_pos = bin)

plot_o_2012 <- ggplot(coverage_df_o_2012, aes(x = start, y = coverage, color = chromosome)) +
    geom_point(alpha = 0.4, size = 1) +
  #geom_smooth(method = "loess", span = 0.2, se = FALSE) +  # Apply smoothing
    facet_wrap(~bam, scales = "free_y", ncol = 4) + # free_y makes all plots have logical y axes, without this they are all plotted on the same axis scale and some are too small to see properly
    theme_minimal() +
    labs(x = "Genomic Position",
         y = "Read Depth (log10 scale)") +
         theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
               aspect.ratio = 1) +  # Ensures all facets have the same shape) + 
    scale_color_manual(values = c("red", "blue"))+
    scale_y_log10() + 
    ggtitle("2012")+
  theme(legend.position = "none") +
  geom_vline(data = switch_df, aes(xintercept = start_pos), color = "black", linetype = "dashed")

#plot_o_2012

# make df

switch_df_labeled_O2012 <- bin_dominance %>%
  group_by(bam) %>%
  arrange(bin) %>%
  mutate(prev_chrom = lag(chromosome)) %>%
  filter(chromosome != prev_chrom & !is.na(prev_chrom)) %>%
  rename(start_pos = bin) %>%
  select(bam, start_pos, chromosome)


# 2015

bin_summary <- coverage_df_o_2015 %>%
  mutate(bin = floor(start / bin_width) * bin_width) %>%
  group_by(bam, bin, chromosome) %>%
  summarize(mean_cov = mean(coverage, na.rm = TRUE), .groups = "drop")

bin_dominance <- bin_summary %>%
  group_by(bam, bin) %>%
  slice_max(mean_cov, n = 1, with_ties = FALSE) %>%  
  ungroup()

switch_df <- bin_dominance %>%
  group_by(bam) %>%
  arrange(bin) %>%
  mutate(prev_chrom = lag(chromosome)) %>%
  filter(chromosome != prev_chrom & !is.na(prev_chrom)) %>%
  select(bam, bin) %>%
  rename(start_pos = bin)

plot_o_2015 <- ggplot(coverage_df_o_2015, aes(x = start, y = coverage, color = chromosome)) +
    geom_point(alpha = 0.4, size = 1) +
  #geom_smooth(method = "loess", span = 0.2, se = FALSE) +  # Apply smoothing
    facet_wrap(~bam, scales = "free_y", ncol = 4) + # free_y makes all plots have logical y axes, without this they are all plotted on the same axis scale and some are too small to see properly
    theme_minimal() +
    labs(x = "Genomic Position",
         y = "Read Depth (log10 scale)") +
         theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
               aspect.ratio = 1) +  # Ensures all facets have the same shape) + 
    scale_color_manual(values = c("red", "blue"))+
    scale_y_log10() + 
    ggtitle("2015")+
  theme(legend.position = "none") +
  geom_vline(data = switch_df, aes(xintercept = start_pos), color = "black", linetype = "dashed")

#plot_o_2015


switch_df_labeled_O2015 <- bin_dominance %>%
  group_by(bam) %>%
  arrange(bin) %>%
  mutate(prev_chrom = lag(chromosome)) %>%
  filter(chromosome != prev_chrom & !is.na(prev_chrom)) %>%
  rename(start_pos = bin) %>%
  select(bam, start_pos, chromosome)




# 2016

bin_summary <- coverage_df_o_2016 %>%
  mutate(bin = floor(start / bin_width) * bin_width) %>%
  group_by(bam, bin, chromosome) %>%
  summarize(mean_cov = mean(coverage, na.rm = TRUE), .groups = "drop")

bin_dominance <- bin_summary %>%
  group_by(bam, bin) %>%
  slice_max(mean_cov, n = 1, with_ties = FALSE) %>%  
  ungroup()

switch_df <- bin_dominance %>%
  group_by(bam) %>%
  arrange(bin) %>%
  mutate(prev_chrom = lag(chromosome)) %>%
  filter(chromosome != prev_chrom & !is.na(prev_chrom)) %>%
  select(bam, bin) %>%
  rename(start_pos = bin)

plot_o_2016 <- ggplot(coverage_df_o_2016, aes(x = start, y = coverage, color = chromosome)) +
    geom_point(alpha = 0.4, size = 1) +
  #geom_smooth(method = "loess", span = 0.2, se = FALSE) +  # Apply smoothing
    facet_wrap(~bam, scales = "free_y", ncol = 4) + # free_y makes all plots have logical y axes, without this they are all plotted on the same axis scale and some are too small to see properly
    theme_minimal() +
    labs(x = "Genomic Position",
         y = "Read Depth (log10 scale)") +
         theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
               aspect.ratio = 1) +  # Ensures all facets have the same shape) + 
    scale_color_manual(values = c("red", "blue"))+
    scale_y_log10() + 
    ggtitle("2016")+
  theme(legend.position = "none") +
  geom_vline(data = switch_df, aes(xintercept = start_pos), color = "black", linetype = "dashed")

#plot_o_2016


switch_df_labeled_O2016 <- bin_dominance %>%
  group_by(bam) %>%
  arrange(bin) %>%
  mutate(prev_chrom = lag(chromosome)) %>%
  filter(chromosome != prev_chrom & !is.na(prev_chrom)) %>%
  rename(start_pos = bin) %>%
  select(bam, start_pos, chromosome)

```






```{r}

oahu_old <- (plot_o_2009 / plot_o_2012 / plot_o_2015 / plot_o_2016) + 
  plot_layout(ncol = 1, heights = c(2, 2, 2, 2), widths = c(2, 4, 3, 3)) #, guides = "collect")  # Adjust heights based on facet count

#old_cov <- ggarrange(plot_2009a, plot_2012a, plot_2015a, plot_2016a, ncol = 1, nrow = 4)

oahu_old

#oahu_all <- (plot_o_2009 / plot_o_2012 / plot_o_2015 / plot_o_2016 / plot_o_2025a) + 
  plot_layout(ncol = 1, heights = c(2, 2, 2, 2, 8), widths = c(2, 4, 3, 3, 4))

#oahu_all
#ggsave("13Jun_oahu_old_cov_plots.pdf", plot = oahu_old, width = 12, height = 12 )


```

```{r}

# 2025

plot_o_2025a <- ggplot(coverage_df_O_2025, aes(x = start, y = coverage, color = chromosome)) +
    geom_line() +
    facet_wrap(~bam, scales = "free_y", ncol = 4) +
    theme_minimal() +
    labs(x = "Genomic Position",
         y = "Read Depth") +
         theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
               aspect.ratio = 1) + 
    scale_color_manual(values = c("red", "blue"))+
  ggtitle("Oahu 2025")+
  theme(legend.position = "none") +
  geom_vline(data = switch_df_labeled_O2025, aes(xintercept = start_pos), color = "black", linetype = "dashed")

plot_o_2025a


#ggsave("26Jun_oahu_2025_cov_plots_v2.pdf", plot = plot_o_2025a, width = 12, height = 12 )


```





## Big Island plots

```{r}

# only data from 2 previous years from BI

# 2012

bin_summary <- coverage_df_BI_2012 %>%
  mutate(bin = floor(start / bin_width) * bin_width) %>%
  group_by(bam, bin, chromosome) %>%
  summarize(mean_cov = mean(coverage, na.rm = TRUE), .groups = "drop")

bin_dominance <- bin_summary %>%
  group_by(bam, bin) %>%
  slice_max(mean_cov, n = 1, with_ties = FALSE) %>%  
  ungroup()

switch_df <- bin_dominance %>%
  group_by(bam) %>%
  arrange(bin) %>%
  mutate(prev_chrom = lag(chromosome)) %>%
  filter(chromosome != prev_chrom & !is.na(prev_chrom)) %>%
  select(bam, bin) %>%
  rename(start_pos = bin)

plot_BI_2012 <- ggplot(coverage_df_BI_2012, aes(x = start, y = coverage, color = chromosome)) +
    geom_point(alpha = 0.4, size = 1) +
  #geom_smooth(method = "loess", span = 0.2, se = FALSE) +  # Apply smoothing
    facet_wrap(~bam, scales = "free_y", ncol = 4) + # free_y makes all plots have logical y axes, without this they are all plotted on the same axis scale and some are too small to see properly
    theme_minimal() +
    labs(x = "Genomic Position",
         y = "Read Depth (log10 scale)") +
         theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
               aspect.ratio = 1) +  # Ensures all facets have the same shape) + 
    scale_color_manual(values = c("red", "blue"))+
    scale_y_log10() + 
    ggtitle("2012")+
  theme(legend.position = "none") +
  geom_vline(data = switch_df, aes(xintercept = start_pos), color = "black", linetype = "dashed")

# plot_BI_2012

# make df

switch_df_labeled_BI2012 <- bin_dominance %>%
  group_by(bam) %>%
  arrange(bin) %>%
  mutate(prev_chrom = lag(chromosome)) %>%
  filter(chromosome != prev_chrom & !is.na(prev_chrom)) %>%
  rename(start_pos = bin) %>%
  select(bam, start_pos, chromosome)



# 2015

bin_summary <- coverage_df_BI_2015 %>%
  mutate(bin = floor(start / bin_width) * bin_width) %>%
  group_by(bam, bin, chromosome) %>%
  summarize(mean_cov = mean(coverage, na.rm = TRUE), .groups = "drop")

bin_dominance <- bin_summary %>%
  group_by(bam, bin) %>%
  slice_max(mean_cov, n = 1, with_ties = FALSE) %>%  
  ungroup()

switch_df <- bin_dominance %>%
  group_by(bam) %>%
  arrange(bin) %>%
  mutate(prev_chrom = lag(chromosome)) %>%
  filter(chromosome != prev_chrom & !is.na(prev_chrom)) %>%
  select(bam, bin) %>%
  rename(start_pos = bin)

plot_BI_2015 <- ggplot(coverage_df_BI_2015, aes(x = start, y = coverage, color = chromosome)) +
    geom_point(alpha = 0.4, size = 1) +
  #geom_smooth(method = "loess", span = 0.2, se = FALSE) +  # Apply smoothing
    facet_wrap(~bam, scales = "free_y", ncol = 4) + # free_y makes all plots have logical y axes, without this they are all plotted on the same axis scale and some are too small to see properly
    theme_minimal() +
    labs(x = "Genomic Position",
         y = "Read Depth (log10 scale)") +
         theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
               aspect.ratio = 1) +  # Ensures all facets have the same shape) + 
    scale_color_manual(values = c("red", "blue"))+
    scale_y_log10() + 
    ggtitle("2015")+
  theme(legend.position = "none") +
  geom_vline(data = switch_df, aes(xintercept = start_pos), color = "black", linetype = "dashed")

# plot_BI_2015

# make df

switch_df_labeled_BI2015 <- bin_dominance %>%
  group_by(bam) %>%
  arrange(bin) %>%
  mutate(prev_chrom = lag(chromosome)) %>%
  filter(chromosome != prev_chrom & !is.na(prev_chrom)) %>%
  rename(start_pos = bin) %>%
  select(bam, start_pos, chromosome)




```

```{r}

BI_old <- (plot_BI_2012 / plot_BI_2015) + 
  plot_layout(ncol = 1, heights = c(2, 2), widths = c(4, 1)) #, guides = "collect")  # Adjust heights based on facet count

#old_cov <- ggarrange(plot_2009a, plot_2012a, plot_2015a, plot_2016a, ncol = 1, nrow = 4)



BI_old

#ggsave("13Jun_BI_old_cov_plots.pdf", plot = BI_old, width = 12, height = 6 )


```

```{r}

# 2025

plot_BI_2025a <- ggplot(coverage_df_BI_2025, aes(x = start, y = coverage, color = chromosome)) +
    geom_line() +
    facet_wrap(~bam, scales = "free_y", ncol = 4) +
    theme_minimal() +
    labs(x = "Genomic Position",
         y = "Read Depth") +
         theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
               aspect.ratio = 1) + 
    scale_color_manual(values = c("red", "blue"))+
  ggtitle("BI 2025")+
  theme(legend.position = "none")+
  geom_vline(data = switch_df_labeled_BI2025, aes(xintercept = start_pos), color = "black", linetype = "dashed")

plot_BI_2025a

#ggsave("26Jun_BI_2025_cov_plots_v2.pdf", plot = plot_BI_2025a, width = 12, height = 6 )


```


All year/island combinations have different numbers of samples, so for simplicity will combine all 4 combo plots into one figure in Illustrator.

Now combine breakpoint dfs into one so I can make schematic


```{r}


switch_df_all <- switch_df_labeled_BI2012 %>%
  bind_rows(switch_df_labeled_BI2015) %>%
  bind_rows(switch_df_labeled_BI2025) %>%
  bind_rows(switch_df_labeled_O2009) %>%
  bind_rows(switch_df_labeled_O2012) %>%
  bind_rows(switch_df_labeled_O2015) %>%
  bind_rows(switch_df_labeled_O2016) %>%
  bind_rows(switch_df_labeled_O2025) 
 
# write.csv(switch_df_all, "switch_df_all.csv")

# now double check and manually curate into df for recomb schematic

```


